redis-hbo-provider/src/main/java/com/facebook/presto/statistic/RedisPlanStatisticsProvider.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statistic;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spi.plan.PlanNodeWithHash;
+import com.facebook.presto.spi.statistics.HistoricalPlanStatistics;
+import com.facebook.presto.spi.statistics.HistoryBasedPlanStatisticsProvider;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.lettuce.core.LettuceFutures;
+import io.lettuce.core.RedisFuture;
+import io.lettuce.core.cluster.api.async.RedisClusterAsyncCommands;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RedisPlanStatisticsProvider
+        implements HistoryBasedPlanStatisticsProvider
+{
+    private static final Logger log = Logger.get(RedisPlanStatisticsProvider.class);
+
+    private final RedisProviderApiStats redisProviderApiStats;
+    private RedisClusterAsyncCommands<String, HistoricalPlanStatistics> commands;
+
+    private final long totalFetchTimeoutMillis;
+
+    private final long totalSetTimeMillis;
+
+    private final long defaultTTLSeconds;
+
+    public RedisPlanStatisticsProvider(RedisClusterAsyncCommands<String, HistoricalPlanStatistics> commands, long totalFetchTimeoutMillis, long totalSetTimeoutMillis, long defaultTTLSeconds)
```

Comment:
@Inject ?

Reply:
Please re-review, I updated the initialization flow